examples/CMakeLists.txt

```diff
@@ -9,4 +9,16 @@ add_custom_target(install_libs
 add_subdirectory(bsoncxx)
 add_subdirectory(mongocxx)
 
-add_custom_target(examples DEPENDS ${MONGOCXX_EXAMPLES} ${BSONCXX_EXAMPLES})
+add_custom_target(examples DEPENDS ${MONGOCXX_EXAMPLE_EXECUTABLES} ${BSONCXX_EXAMPLE_EXECUTABLES})
+
+# Run all mongocxx examples on `make examples`.
+foreach(EXAMPLE ${MONGOCXX_EXAMPLE_EXECUTABLES})
+    get_filename_component(EXAMPLE_EXECUTABLE "${CMAKE_BINARY_DIR}/examples/mongocxx/${EXAMPLE}" ABSOLUTE)
+    add_custom_command(TARGET examples POST_BUILD COMMAND ${EXAMPLE_EXECUTABLE})
```

Comment:
Lets call the target 'run-examples'. It makes it clear that you are going to run something, not just build it.